core/services/ocrcommon/peer_wrapper.go

```diff
@@ -0,0 +1,228 @@
+			}
+
+			peerStore = p.pstoreWrapper.Peerstore
+			announcePort = p.config.P2PListenPort()
+			if p.config.P2PAnnouncePort() != 0 {
+				announcePort = p.config.P2PAnnouncePort()
+			}
+			announceIP = p.config.P2PListenIP()
+			if p.config.P2PAnnounceIP() != nil {
+				announceIP = p.config.P2PAnnounceIP()
+			}
+		}
+
+		// Discover DB is only required for v2
+		// Also fallback to listen addresses if announce not specified
+		var discovererDB ocrnetworkingtypes.DiscovererDatabase
+		var announceAddresses []string
+		if ns == ocrnetworking.NetworkingStackV2 || ns == ocrnetworking.NetworkingStackV1V2 {
+			discovererDB = NewDiscovererDatabase(p.db.DB, p2ppeer.ID(p.PeerID))
+			announceAddresses = p.config.P2PV2ListenAddresses()
+			if len(p.config.P2PV2AnnounceAddresses()) != 0 {
+				announceAddresses = p.config.P2PV2AnnounceAddresses()
+			}
+		}
+
+		peerConfig := ocrnetworking.PeerConfig{
+			NetworkingStack: p.config.P2PNetworkingStack(),
+			PrivKey:         key.PrivKey,
+			Logger:          logger.NewOCRWrapper(p.lggr, p.config.OCRTraceLogging(), func(string) {}),
+
+			// V1 config
+			V1ListenIP:                         p.config.P2PListenIP(),
+			V1ListenPort:                       p.config.P2PListenPort(),
+			V1AnnounceIP:                       announceIP,
+			V1AnnouncePort:                     announcePort,
+			V1Peerstore:                        peerStore,
+			V1DHTAnnouncementCounterUserPrefix: p.config.P2PDHTAnnouncementCounterUserPrefix(),
+
+			// V2 config
+			V2ListenAddresses:    p.config.P2PV2ListenAddresses(),
+			V2AnnounceAddresses:  announceAddresses,
+			V2DeltaReconcile:     p.config.P2PV2DeltaReconcile().Duration(),
+			V2DeltaDial:          p.config.P2PV2DeltaDial().Duration(),
+			V2DiscovererDatabase: discovererDB,
+
+			EndpointConfig: ocrnetworking.EndpointConfig{
+				// V1 and V2 config
+				IncomingMessageBufferSize: p.config.P2PIncomingMessageBufferSize(),
+				OutgoingMessageBufferSize: p.config.P2POutgoingMessageBufferSize(),
+
+				// V1 Config
+				NewStreamTimeout:       p.config.P2PNewStreamTimeout(),
+				DHTLookupInterval:      p.config.P2PDHTLookupInterval(),
+				BootstrapCheckInterval: p.config.P2PBootstrapCheckInterval(),
+			},
+		}
+
+		p.lggr.Debugw("Creating OCR/OCR2 Peer", "config", peerConfig)
+		peer, err := ocrnetworking.NewPeer(peerConfig)
+		if err != nil {
+			return errors.Wrap(err, "error calling NewPeer")
+		}
+		p.Peer = &peerAdapter{
+			peer,
+			peer.OCR1BinaryNetworkEndpointFactory(),
+			peer.OCR1BootstrapperFactory(),
+		}
+		p.Peer2 = &peerAdapter2{
+			peer,
+			peer.OCR2BinaryNetworkEndpointFactory(),
+			peer.OCR2BootstrapperFactory(),
+		}
+		return nil
+	})
+}
+
+// Close closes the peer and peerstore
+func (p *SingletonPeerWrapper) Close() error {
+	return p.StopOnce("SingletonPeerWrapper", func() (err error) {
+		if p.Peer != nil {
+			err = p.Peer.Close()
+		}
+		if p.Peer2 != nil {
+			err = p.Peer2.Close()
+		}
```

Comment:
Maybe we want to combine these since I think its possible we can have both Peer and Peer2 in the case of V1V2 networking

Reply:
no not possible - Peer is required for the ocr1 oracle instantiation and Peer2 is required for ocr2 oracle instantiation